### ISSUE TITLE
Low loss but bad results even on training data

### ISSUE THREAD

**Issue by arslanan** [facebookresearch/detectron2#120]
Hello, 

After following the tutorial, I tried to train a model with a custom dataset. 
The model trains fine and the loss seems to reduce a lot (for 300 iterations it goes below 0.1)
 
`eta: 0:00:03  iter: 99  total_loss: 0.156  loss_cls: 0.102  loss_box_reg: 0.000  loss_rpn_cls: 0.036  loss_rpn_loc: 0.012 `
(exemple for 99 iter) 

But whe I try to use the model (still folowing the tutorial) only a few items are detected with a very low confidence (<10%) 
Nevertheless, there are almost no wrong item detected. 

Plus, item are detected in multiple bounding box as if they were no NMS. 

Can you tell me if I missed something in the implementation ? 

Here is the implementation : 

`cfg = get_cfg()` 
`cfg.merge_from_file("./detectron2_repo/configs/COCO-Detection/faster_rcnn_X_101_32x8d_FPN_3x.yaml")` 
` cfg.DATASETS.TRAIN = ("train",)` 
` cfg.DATASETS.TEST = ()  ` 
` cfg.DATALOADER.NUM_WORKERS = 2` 
` cfg.MODEL.WEIGHTS = "detectron2://COCO-Detection/faster_rcnn_X_101_32x8d_FPN_3x/139173657/model_final_68b088.pkl"  ` 
` cfg.SOLVER.IMS_PER_BATCH = 2` 
` cfg.SOLVER.BASE_LR = 0.00025` 
` cfg.SOLVER.MAX_ITER = 300` 
` cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 128 ` 
` cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1  # only has one class (paraphe)` 
` cfg.SOLVER.CHECKPOINT_PERIOD = 50` 
` cfg.MODEL.ANCHOR_GENERATOR.SIZES=[[20],[12],[32],[40],[17]]` 
` cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1`

**Comment by ppwwyyxx** [comment]
You may have better luck starting with the an existing working config and train for more iterations than 300.  But unfortunately we do not help users train models on their datasets. As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) says:
> If you expect the model to work better, only in one of the two conditions we will help with it: (1) You're unable to reproduce the results documented in detectron2 model zoo. (2) It indicates a detectron2 bug.
